File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        // [2,7,11,15], target = 9

        for(int i = 0; i < nums.length; i++) { // 2 // 7
            if(map.containsKey(target - nums[i])) { // 9 - 2 -> 7 // 9 - 7 -> 2

                return new int[]{map.get(target-nums[i]), i}; // 1, 0
            }
            map.put(nums[i], i); // 2 : 0, 
        }

        return new int[]{-1, -1};
    }
}

File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> map = new Dictionary<int, int>();

        for(int i = 0; i < nums.Length; i++) {
            if(map.ContainsKey(target - nums[i])) {
                return new int[] {map[target - nums[i]], i};
            }
            map[nums[i]] = i;
        }
        return new int[] {-1, -1};
    }
}